packages/flutter/lib/src/cupertino/date_picker.dart

```diff
@@ -413,6 +418,9 @@ class CupertinoDatePicker extends StatefulWidget {
   /// Defaults to a value that matches the default iOS date picker wheel.
   final double itemExtent;
 
+  /// A builder to return the selection overlay widget for the picker.
```

Comment:
I think this needs to be a bit more detailed. An explanation on how to recreate the default behavior manually with this builder should be enough. From there, I think developers can figure out how to do their use case.

So some mention of how `CupertinoPickerDefaultSelectionOverlay` is used by default. And if we allow the builder to return null, a mention about being able to do that to have no overlay would be good.